File: MovieProject/MovieProject.Web/Modules/Northwind/OrderDetailsExtended/OrderDetailsExtendedColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.OrderDetailsExtended")]
[BasedOnRow(typeof(OrderDetailsExtendedRow), CheckNames = true)]
public class OrderDetailsExtendedColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int OrderId { get; set; }
    public int ProductId { get; set; }
    [EditLink]
    public string ProductName { get; set; }
    public decimal UnitPrice { get; set; }
    public short Quantity { get; set; }
    public float Discount { get; set; }
    public decimal ExtendedPrice { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrderDetailsExtendedColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDetailsExtendedRow } from "./OrderDetailsExtendedRow";

export interface OrderDetailsExtendedColumns {
    OrderId: Column<OrderDetailsExtendedRow>;
    ProductId: Column<OrderDetailsExtendedRow>;
    ProductName: Column<OrderDetailsExtendedRow>;
    UnitPrice: Column<OrderDetailsExtendedRow>;
    Quantity: Column<OrderDetailsExtendedRow>;
    Discount: Column<OrderDetailsExtendedRow>;
    ExtendedPrice: Column<OrderDetailsExtendedRow>;
}

export class OrderDetailsExtendedColumns extends ColumnsBase<OrderDetailsExtendedRow> {
    static readonly columnsKey = 'Northwind.OrderDetailsExtended';
    static readonly Fields = fieldsProxy<OrderDetailsExtendedColumns>();
}